linked_list.py

```diff
@@ -21,18 +25,20 @@ def pop(self):
         try:
             head_val = self.head.val
             self.head = self.head.next
+            self.list_size -= 1
             return head_val
         except AttributeError:
-            return "No head"
+            raise ValueError(u"No head in this list.")
```

Comment:
letting your users see 'No head' might be more of an internal implementation detail.  Might be better just to tell them it's empty.